src/translib/transformer/xfmr_protocols.go

```diff
@@ -0,0 +1,51 @@
+package transformer
+
+import (
+    "strings"
+    "translib/db"
+    log "github.com/golang/glog"
+)
+
+func init () {
+    XlateFuncBind("network_instance_protocols_ptotocol_table_name_xfmr", network_instance_protocols_ptotocol_table_name_xfmr)
+    XlateFuncBind("YangToDb_network_instance_protocol_key_xfmr", YangToDb_network_instance_protocol_key_xfmr)
+    XlateFuncBind("DbToYang_network_instance_protocol_key_xfmr", DbToYang_network_instance_protocol_key_xfmr)
+
+}
+
+var network_instance_protocols_ptotocol_table_name_xfmr TableXfmrFunc = func (inParams XfmrParams)  ([]string, error) {
+    var tblList []string
+
+    log.Info("network_instance_protocols_protocol_table_name_xfmr")
+    if (inParams.oper == GET) {
+        if(inParams.dbDataMap != nil) {
+            (*inParams.dbDataMap)[db.ConfigDB]["CFG_PROTO_TBL"] = make(map[string]db.Value)
+            (*inParams.dbDataMap)[db.ConfigDB]["CFG_PROTO_TBL"]["BGP|bgp"] = db.Value{Field: make(map[string]string)}
+            (*inParams.dbDataMap)[db.ConfigDB]["CFG_PROTO_TBL"]["BGP|bgp"].Field["NULL"] = "NULL"
```

Comment:
Assuming "bgp" for protocol name does not look good, as per OC, user can give any name and supporting BGP and other protocols under protocols container without configuring the protocols list seems to be deviating the OC compliance itself, either store the protocols in the config-DB or allow to cache some information in the transformer to supply information like this to the OC query when it is not actually available in the DB.